File: src/main/java/com/blink/backend/controller/appointment/dto/UpdateAppointmentDTO.java
package com.blink.backend.controller.appointment.dto;

import com.blink.backend.domain.model.AppointmentUpdate;
import com.blink.backend.persistence.entity.appointment.AppointmentStatus;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class UpdateAppointmentDTO {
    private String notes;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm")
    private LocalDateTime scheduledTime;
    private AppointmentStatus status;

    public AppointmentUpdate toDomain() {
        return new AppointmentUpdate(
                notes,
                scheduledTime,
                status
        );
    }
}

File: src/main/java/com/blink/backend/controller/appointment/dto/UpdateAppointmentDto.kt
package com.blink.backend.controller.appointment.dto

import com.blink.backend.domain.model.AppointmentUpdate
import com.blink.backend.persistence.entity.appointment.AppointmentStatus
import java.time.LocalDateTime

data class UpdateAppointmentDto(
    val notes: String? = null,
    val scheduledTime: LocalDateTime? = null,
    val status: AppointmentStatus? = null,
) {
    fun toDomain(): AppointmentUpdate {
        return AppointmentUpdate(
            notes,
            scheduledTime,
            status
        )
    }
}